### PR TITLE
feat(Communities): Redesigned fees box and sign tx popup components

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

- implements "fees box" and sign tx popup according to the new design, covering all variants
- add small utility to the Storybook
- fixes bug in `ModelUtils.get` potentially causing UB or crash

Closes: https://github.com/status-im/status-desktop/issues/11600

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Screenshot from 2023-07-20 12-48-18
Screenshot from 2023-07-20 12-48-55

### CODE REVIEW

**Review by caybro — APPROVED** [review]
LGTM

**Review by alexandraB99 — APPROVED** [review]

File: ui/app/AppLayouts/Communities/controls/FeeRow.qml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import QtQuick 2.15
```
Comment (alexjba) [inline ui/app/AppLayouts/Communities/controls/FeeRow.qml:1]: I'm getting some warnings on this file. Might be different on other Qt versions. Do you also see them in storybook?
```
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
```

**Review by alexjba — COMMENTED** [review]
Looking good so far!

File: ui/app/AppLayouts/Communities/controls/FeeRow.qml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                elide: Text.ElideRight
+            }
+
+            Item {
+                Layout.preferredWidth: (parent.width - parent.spacing) / 2
+                Layout.alignment: Qt.AlignTop
+
+                implicitHeight: feeText.implicitHeight
+
+                StatusBaseText {
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/app/AppLayouts/Communities/controls/FeeRow.qml:44]: I haven't... But it's great idea, for sure worth checking. Will do, thanks!

**Review by noeliaSD — APPROVED** [review]
LGTM! I didn't test it, only code inspection!

Some minor comments added!

File: ui/app/AppLayouts/Communities/controls/FeeRow.qml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    id: maxText
+                    text: qsTr("Max.")
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/app/AppLayouts/Communities/controls/FeeRow.qml:46]: I think the goal is to inform the user that the actual cost can be lower but for sure not higher then the amount displayed.

File: storybook/pages/FeesBoxPage.qml [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+SplitView {
+    Logs { id: logs }
+
+    FeesModel {
+        id: feesModel
+    }
+
+    ListModel {
+        id: accountsModel
```
[the remark this answers is not in the document]
Comment (micieslak) [reply storybook/pages/FeesBoxPage.qml:20]: I will do in the next PR.

File: ui/app/AppLayouts/Communities/controls/FeeRow.qml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    property bool highlightFee: false
+    property bool errorFee: false
+
+    background: null
+
+    contentItem: RowLayout {
+        StatusBaseText {
+            Layout.fillWidth: true
+            Layout.maximumWidth: (parent.width - parent.spacing) / 2
```
Comment (alexjba) [inline ui/app/AppLayouts/Communities/controls/FeeRow.qml:25]:
```suggestion
           Binding on Layout.maximumWidth {
                value: (layout.width - layout.spacing) / 2
                delayed: true
            }
```

This is fixing the StackLayout warning I'm having.

Comment (alexjba) [reply ui/app/AppLayouts/Communities/controls/FeeRow.qml:25]: False alarm! It's not a proper fix. The layout for FeeRow and FeesPanel is broken

Comment (micieslak) [reply ui/app/AppLayouts/Communities/controls/FeeRow.qml]: Fixed by using Item on toplevel :)

**Review by alexjba — APPROVED** [review]
LGTM! Nice work!

File: ui/app/AppLayouts/Communities/controls/FeeRow.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Communities/controls/FeeRow.qml:1]: solved!

**Pull request merged**